### PR TITLE
React Native <-> WebView communication support for Android 

### LINKED ISSUE #14: React Native <-> WebView communication support for Android
Same as https://github.com/software-mansion-labs/react-native-turbo-demo/issues/9 but for Android.

### PR DESCRIPTION
**Pull request by Fryzu**
Fixes #14 

## Notes: 
- android doesn't support async code response from WebView yet

### CODE REVIEW

File: android/app/src/main/java/com/hotwirerndemo/Utils.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+                        else -> bundle.putString(key, value.javaClass.simpleName)
+                    }
+                }
+            } catch (e: JSONException) {
+                e.printStackTrace()
+            }
+            return bundle
+        }
+    }
+}
```
Comment (lukmccall) [inline android/app/src/main/java/com/hotwirerndemo/Utils.kt:33]:
```suggestion
}

```

File: android/app/src/main/java/com/hotwirerndemo/Utils.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+                        "Boolean" -> bundle.putBoolean(key, value as Boolean)
+                        "JSONObject" -> bundle.putBundle(key, convertJsonToBundle(value as JSONObject))
+                        "Float" -> bundle.putFloat(key, value as Float)
+                        "Double" -> bundle.putDouble(key, value as Double)
+                        else -> bundle.putString(key, value.javaClass.simpleName)
+                    }
+                }
+            } catch (e: JSONException) {
+                e.printStackTrace()
+            }
```
Comment (lukmccall) [inline android/app/src/main/java/com/hotwirerndemo/Utils.kt:29]: Maybe it will be better to throw that error and crash the application. What do you think?

**Review by lukmccall — APPROVED** [review]

File: ios/RNVisitableView.swift [hunk 3]
```diff
@@ -68,7 +68,7 @@ extension RNVisitableView: RNVisitableViewControllerDelegate {
 
   func visitableWillAppear(visitable: Visitable) {
     print("View will appear for URL", visitable.visitableURL.absoluteString)
-    let session = getSession()
+    let session = getSession()
```
Comment (pfeiffer) [inline ios/RNVisitableView.swift:71]: Unneeded whitespace

File: android/app/src/main/java/com/hotwirerndemo/RNSession.kt [hunk 4]
```diff
@@ -1,27 +1,36 @@
...
-import dev.hotwire.turbo.visit.TurboVisit
 import java.util.*
+import org.json.JSONObject
 
 class RNSession(context: Context) : FrameLayout(context) {
 
+    private val reactContext = context as ReactContext
+    private val registeredVisitableViews = mutableListOf<SessionSubscriber>()
+
     val session: TurboSession = run {
```
Comment (Fryzu) [inline android/app/src/main/java/com/hotwirerndemo/RNSession.kt:21]: Rename to turboSession https://github.com/software-mansion-labs/react-native-turbo-demo/pull/15#discussion_r993119067

**Pull request merged**